### PR TITLE
[vcpkg baseline][ocilib] Fix build failure

### PR DESCRIPTION
**Pull request by JackBoosY**
The target architecture in different sln is different.

Related: #11637.

### CODE REVIEW

File: ports/ocilib/portfile.cmake [hunk 1]
```diff
@@ -13,11 +13,23 @@ vcpkg_from_github(
 if(VCPKG_TARGET_IS_WINDOWS)
 	if(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
 		set(SOLUTION_TYPE vs2019)
```
Comment (LilyWangL) [inline ports/ocilib/portfile.cmake:15]:
```suggestion
set(SOLUTION_TYPE vs2019)
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/ocilib/portfile.cmake [hunk 2]
```diff
@@ -13,11 +13,23 @@ vcpkg_from_github(
...
+        set(OCILIB_ARCH_X86 x86)
+        set(OCILIB_ARCH_X64 x64)
 	elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
 		set(SOLUTION_TYPE vs2017)
```
Comment (LilyWangL) [inline ports/ocilib/portfile.cmake:19]:
```suggestion
  set(SOLUTION_TYPE vs2017)
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/ocilib/portfile.cmake [hunk 3]
```diff
@@ -13,11 +13,23 @@ vcpkg_from_github(
 if(VCPKG_TARGET_IS_WINDOWS)
 	if(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
 		set(SOLUTION_TYPE vs2019)
+        set(OCILIB_ARCH_X86 x86)
+        set(OCILIB_ARCH_X64 x64)
 	elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
-		set(SOLUTION_TYPE vs2017)
+  set(SOLUTION_TYPE vs2017)
```
Comment (LilyWangL) [inline ports/ocilib/portfile.cmake:19]:
```suggestion
        set(SOLUTION_TYPE vs2017)
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/ocilib/portfile.cmake [hunk 4]
```diff
@@ -12,12 +12,24 @@ vcpkg_from_github(
...
+        set(SOLUTION_TYPE vs2019)
+        set(OCILIB_ARCH_X86 x86)
+        set(OCILIB_ARCH_X64 x64)
 	elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
-		set(SOLUTION_TYPE vs2017)
+        set(SOLUTION_TYPE vs2017)
+        set(OCILIB_ARCH_X86 Win32)
+        set(OCILIB_ARCH_X64 Win64)
 	else()
 		set(SOLUTION_TYPE vs2015)
```
Comment (LilyWangL) [inline ports/ocilib/portfile.cmake:23]:
```suggestion
        set(SOLUTION_TYPE vs2015)
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Review by LilyWangL — APPROVED** [review]

**Comment by BillyONeal** [comment]
This passes locally for me, what drove this change?

**Comment by JackBoosY** [comment]
@BillyONeal This is indeed a port bug, it seems that the pipeline did not catch this error correctly.

**Comment by BillyONeal** [comment]
@JackBoosY I believe that it is a port bug, but it passes in CI and it passed for me locally, so I was hoping to have a more concrete analysis on why. Oh well, no problem, merged now :)

**Pull request merged**